quarkus/addons/knative/serving/runtime/src/main/java/org/kie/kogito/addons/quarkus/knative/serving/customfunctions/CloudEventKnativeServiceRequestClient.java

```diff
@@ -79,30 +84,20 @@ protected JsonNode sendRequest(String processInstanceId, URI serviceAddress, Str
         return responseAsJsonObject(response);
     }
 
-    private static HashMap<String, Object> createCloudEventWithGeneratedId(Map<String, Object> cloudEvent, String processInstanceId) {
-        HashMap<String, Object> modifiableCloudEvent = new HashMap<>(cloudEvent);
-        Object oldIdValue = modifiableCloudEvent.put("id", generateCloudEventId(processInstanceId, cloudEvent.get("source").toString()));
-
-        if (oldIdValue != null) {
-            logger.warn("CloudEvent id should not be set. Ignoring id {}. A generated value will be used instead.", oldIdValue);
+    private static Map<String, Object> createCloudEventWithGeneratedId(Map<String, Object> cloudEvent, String processInstanceId) {
+        Map<String, Object> modifiableCloudEvent = new HashMap<>(cloudEvent);
+        Object source = cloudEvent.get("source");
+        if (source == null) {
+            throw new IllegalArgumentException("Invalid CloudEvent: Attribute source is mandatory.");
         }
+        modifiableCloudEvent.put(ID, generateCloudEventId(processInstanceId, source.toString()));
         return modifiableCloudEvent;
     }
```

Comment:
I think there is still issues with source, so here is my proposal to generate an unique id with processinsanceid

```suggestion
    private static Map<String, Object> createCloudEventWithGeneratedId(Map<String, Object> cloudEvent, String processInstanceId) {
        Map<String, Object> modifiableCloudEvent =ensureModifiable(cloudEvent);
        modifiableCloudEvent.put(ID, processInstanceId + "_" + cloudEvent.hashCode());
        return modifiableCloudEvent;
    }
```

Reply:
Together with
```
private static Map<String,Object> ensureModifiable (Map<String,Object> map) {
     return map instanceof HashMap ? map : new HashMap(map);
}
```

Reply:
Done.